tlscanary/template/js/index_page.js

```diff
@@ -0,0 +1,346 @@
+  var columnData = $("#grid").bootgrid("getColumnSettings");
+  var columns = [];
+  for (var i = 0;i < columnData.length;i++) {
+      if (columnData[i].visible)
+      {
+        if (columnData[i].id !== "Actions")
+        {
+          columns.push (columnData[i].id);
+        }
+      }
+  }
+  return columns;
+}
+
+function getSortedRows(fieldName) {
+  var gridData = $("#grid").bootgrid().data(".rs.jquery.bootgrid").rows;
+  var currentRows = [];
+  var searchStr = $("#grid").bootgrid("getSearchPhrase");
+  var currentColumns = getVisibleColumns();
+  if (fieldName == "All")
+  {
+    fieldName = "";
+  }
+  for (var i = 0;i < gridData.length;i++) {
+    var row = gridData[i];
+    for (var j = 0;j < currentColumns.length;j++)
+    {
+      var field = row[currentColumns[j]].toString();
+      if (field.search(searchStr) != -1 && field.search(fieldName) != -1)
+      {
+        currentRows.push(row);
+        break;
+      }
+    }
+  }
+  return currentRows;
+}
+
+function makeTable(hosts, columns) {
+  // First, add new column for our grid actions
+  columns.push(
+    {
+      name: "Actions",
+      default: true,
+      type: null,
+      width: "20%"
+    }
+  );
+  // Add an invisible column which allows one field to become the identifier
+  columns.unshift(
+    {
+      name: "id",
+      default: false,
+      type: "str",
+      width: "0%"
+    }
+  );
+
+  var html = "<table id='grid' class='table table-condensed table-hover table-striped'><thead><tr>";
+  for (var i = 0;i < columns.length;i++) {
+    html += "<th data-column-id='" + columns[i].name + "' ";
+    if (columns[i].name === "id")
+    {
+      html += "data-visible-in-selection='false'  data-searchable='false' ";
+      html += "data-identifier='true' ";
+    } 
+    if (columns[i].name === "Actions")
+    {
+      html += "data-visible-in-selection='false' data-formatter='commands' data-searchable='false' ";
+    }
+    if (columns[i].name === "run")
+    {
+      html += "data-order='desc' data-formatter='link' "; 
+    }
+    if (!columns[i].default)
+    {
+      html += "data-visible='false' ";
+    }
+    if (columns[i].type === "int")
+    {
+      html += "data-type='numeric' ";
+    }
+    if (typeof(columns[i].width) !== "undefined")
+    {
+      html += "data-width='" + columns[i].width + "' ";
+    } else {
+      html += "data-width='20%' ";
+    }
+    html += ">" + columns[i].name + "</th>"
+  }
+  html += "</tr></thead><tbody>";
+  for (var i = 0;i < hosts.length;i++) {
+    html += "<tr id=\'" + hosts[i]["run"] + "\'>";
+    html += "<td>" + hosts[i]["run"] + "</td>"
+    for (var j = 1;j < columns.length;j++)
+    {
+      html += "<td>" + hosts[i][columns[j].name] + "</td>"
+    }
+    html += "</tr>";
+  }
+  html += "</tbody></table>";
+  const contentDiv = document.getElementById("results");
+  contentDiv.style.visibility = "hidden";
+  contentDiv.innerHTML = html;
+}
+
+function applyBootgrid() {
+  const grid = $("#grid").bootgrid(
+  {
```

Comment:
Stomach says this curly should trail the previous line without linebreak. What do you think?